semantic/src/main/kotlin/tools/samt/semantic/SemanticChecker.kt

```diff
@@ -0,0 +1,31 @@
+package tools.samt.semantic
+
+import tools.samt.common.DiagnosticConsole
+import tools.samt.parser.FileNode
+import tools.samt.semantic.checks.*
+
+internal interface SemanticCheck {
+    fun check(fileNode: FileNode)
+}
+
+class SemanticChecker private constructor(private val fileNode: FileNode, diagnostics: DiagnosticConsole) {
```

Comment:
In regard to all checks currently implemented: Should we rename these to indicate that they are performed before any type information is present? I assume future checks will also be implemented as SemanticChecker passes, how will we separate them from the ones that run before any type info is present?

Reply:
I would've called the checks after the type system has run `TypeChecker` and so on but am open to suggestions if you have a more self-describing name for this iteration of `SemanticChecker` without type information